src/app/components/product-table/product-table.component.html

```diff
@@ -1,5 +1,11 @@
 <mat-grid-list cols="3" rowHeight="400px">
-    <mat-grid-tile *ngFor="let product of products">
-        <app-product-card [product]="product"></app-product-card>
-    </mat-grid-tile>
+    <ng-container *ngIf="products$ | async as products; else spinner">
```

Comment:
Тоже бы чуток подправил, обрати внимание на подобные места
```suggestion
    <ng-container *ngIf="(products$ | async) as products; else spinner">
```